internal/patroni/config_test.go

```diff
@@ -250,13 +249,17 @@ func TestDynamicConfiguration(t *testing.T) {
 		},
 		{
 			name: "top-level passes through",
-			input: map[string]any{
-				"retry_timeout": 5,
+			spec: v1beta1.PostgresClusterSpec{
+				Patroni: &v1beta1.PatroniSpec{
+					DynamicConfiguration: map[string]any{
+						"retry_timeout": int64(5),
```

Comment:
:thinking: Why do we need `int64` here and below?

Reply:
The `DeepCopy()` function is only happy with types produced by the `json` package plus `int64`. When parsed, these numbers are `float64`. 🤔 Maybe I should use that type. Another approach could be YAML instead of structs here.

```go
	var object map[string]any
	fmt.Println(yaml.Unmarshal([]byte(`{
		asdf: 1,
	}`), &object))

	fmt.Printf("%#v\n", object)
	fmt.Printf("%T\n", object["asdf"])
```

```
<nil>
map[string]interface {}{"asdf":1}
float64
```

Reply:
I pushed another commit that unmarshals yaml. WDYT?

Reply:
Maybe `float64` would make sense, but I'm also fine with leaving this as is. I mainly wanted to understand the reason for the type change and now I do :+1:

Reply:
The new changes work for me, perhaps it's even a little more decipherable?